dvc/repo/metrics/show.py

```diff
@@ -171,20 +172,28 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
             - xpath:
     """
     outs = [out for stage in repo.stages() for out in stage.outs]
+    found = []
 
-    if path:
+    # Display all metrics if no target is given
+    if not targets:
+        targets = []
+        found = outs
+
+    for target in targets:
         try:
-            outs = repo.find_outs_by_path(path, outs=outs, recursive=recursive)
+            found += repo.find_outs_by_path(
+                target, outs=outs, recursive=recursive
+            )
         except OutputNotFoundError:
             logger.debug(
                 "DVC-file not for found for '{}' in branch '{}'".format(
-                    path, branch
+                    target, branch
                 )
             )
             return []
```

Comment:
Also this no longer makes sense, because you are returning on first failed target, even though you should process everything you can.

Reply:
Maybe it's better to raise the `BadMetricError` instead.

Reply:
Never mind, it halts the metric recollection with `all_branches` / `all_targets`.